### ISSUE TITLE
Add support for --run-after and --run-before to a particular section in coafile

### ISSUE THREAD

**Issue by khanchi97** [coala/coala#5037]
.coafile
```
[a]
setting1...

[b]
setting2...

[c]
setting3...

[d]
setting4...
```

Now coala should be able to run commands 
1.`coala --run-after b`       => coala will only execute section `c` and `d`
2.`coala --run-before c`    => coala will only execute section `a` and `b`

**Comment by ishanSrt** [comment]
would like to take this up

**Comment by khanchi97** [comment]
@ishanSrt first let see what discussion will take it to.Is this behaviour is desired or not?
Then you can work on this :)

**Comment by Makman2** [comment]
Haven't we agreed on this:
```
coala a b c d
coala d b c a
```
That someone has to specify the order like this?

**Comment by khanchi97** [comment]
yeah, we agreed.But what if in `.coafile` there are many sections then user would like to use this feature `--run-after` rather than passing all those sections through cli.

**Comment by Makman2** [comment]
I'm not sure if there's such a use-case. Could we discuss that on a different issue? It's in fact a new feature.

**Comment by khanchi97** [comment]
>Could we discuss that on a different issue?It's in fact a new feature.

No problem :)
>Or we open a new issue for
     `coala a b c d`
     `coala d b c a`

We have already opened https://github.com/coala/coala/issues/5038
and I've made a PR too https://github.com/coala/coala/pull/5068 ;)

**Comment by Makman2** [comment]
> We have already opened #5038

Ah perfect, I already wondered about having read something similar 😅

**Comment by Makman2** [comment]
Okay I'm not sure if this complicates our CLI. If the ordering of bigger coafiles needs to be altered, it usually should be done in the coafile itself.

**Comment by ishanSrt** [comment]
support will have to be added if a user uses both cases i.e. `--run-after ` and then orders the sections accordingly too

**Comment by Makman2** [comment]
Makes it even more complex... I'm not really in for this feature.

@sils @jayvdb opinions?
